app/gui/view/src/project/project_view_top_bar/window_control_buttons.rs

```diff
@@ -0,0 +1,199 @@
+//! The component with buttons in the top left corner. See [[View]].
+
+use ensogl::display::shape::*;
+use ensogl::prelude::*;
+
+use enso_frp as frp;
+use ensogl::application;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::shape;
+use ensogl_hardcoded_theme::application::window_control_buttons as theme;
+
+
+// ==============
+// === Export ===
+// ==============
+
+pub mod close;
+pub mod fullscreen;
+
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+mod shape {
+    use super::*;
+    shape! {
+        alignment = center;
+        (style: Style) {
+            Plane().fill(INVISIBLE_HOVER_COLOR).into()
+        }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// An internal model of Status Bar component
+#[derive(Clone, CloneRef, Debug)]
+pub struct Model {
+    app:            Application,
+    display_object: display::object::Instance,
+    shape:          shape::View,
```

Comment:
Why this shape is needed? It is behind buttons, so looks like it is needed for some mouse interaction, but it is not described here. Also, its name does not explain its purpose. The name "shape" does not semantically tell what it is used for.

Reply:
Old code.